cmd/xl-zones.go

```diff
@@ -1062,6 +1062,37 @@ func (z *xlZones) PutObjectPart(ctx context.Context, bucket, object, uploadID st
 	}
 }
 
+func (z *xlZones) GetMultipartInfo(ctx context.Context, bucket, object, uploadID string, opts ObjectOptions) (MultipartInfo, error) {
+	if err := checkListPartsArgs(ctx, bucket, object, z); err != nil {
+		return MultipartInfo{}, err
+	}
+
+	uploadIDLock := z.NewNSLock(ctx, bucket, pathJoin(object, uploadID))
+	if err := uploadIDLock.GetRLock(globalOperationTimeout); err != nil {
+		return MultipartInfo{}, err
+	}
+	defer uploadIDLock.RUnlock()
+
+	if z.SingleZone() {
+		return z.zones[0].GetMultipartInfo(ctx, bucket, object, uploadID, opts)
+	}
+	for _, zone := range z.zones {
+		result, err := zone.ListMultipartUploads(ctx, bucket, object, "", "", "", maxUploadsList)
```

Comment:
We can avoid the ListMultipartUploads call here. We have everything we need to fetch the corresponding multipartInfo.